### ISSUE TITLE
Rule proposal: no catch throw

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#2466]
### Description

https://x.com/devagrawal09/status/1842073035517403534

Not sure how common this is, but. Maybe it’s an easy rule?

### Fail

```js
try {
  something()
} catch (error) {
  throw error
}
```

```js
something.catch(error => {throw error})
```

### Pass

```js
something()
```

```js
try {
  something()
} catch (error) {
  errors.add(error)
  throw error
}
```

### Proposed rule name

no-useless-catch-throw

### Additional Info

I’d rather see this first though:

- https://github.com/sindresorhus/eslint-plugin-unicorn/issues

**Comment by dimaMachina** [comment]
Eslint already has built-in rule :) https://eslint.org/docs/latest/rules/no-useless-catch

**Comment by fregante** [comment]
Perfect!



> not sure if it works with 2nd failed example

Doesn't look like it, but we have https://github.com/eslint-community/eslint-plugin-promise/blob/main/docs/rules/prefer-await-to-then.md and in the future:

- https://github.com/sindresorhus/eslint-plugin-unicorn/issues/652
